processing/src/main/java/io/druid/query/filter/ExtractionDimFilter.java

```diff
@@ -98,7 +99,12 @@ public DimFilter optimize()
   {
     if (this.getExtractionFn() instanceof LookupExtractionFn
         && ((LookupExtractionFn) this.getExtractionFn()).isOptimize()) {
-      LookupExtractor lookup = ((LookupExtractionFn) this.getExtractionFn()).getLookup();
+      // case ReplaceMissingValueWith matching the value we can not optimize
+      final LookupExtractionFn lookupExtractionFn = (LookupExtractionFn) this.getExtractionFn();
+      if (Strings.emptyToNull(lookupExtractionFn.getReplaceMissingValueWith()) == Strings.emptyToNull(value)) {
```

Comment:
It's fine to leave out the emptyToNull() call on ```Strings.emptyToNull(lookupExtractionFn.getReplaceMissingValueWith()```, since the constructor of FunctionalExtraction already does that:

```
  public FunctionalExtraction(
      final Function<String, String> extractionFunction,
      final boolean retainMissingValue,
      final String replaceMissingValueWith,
      final boolean injective
  )
  {
    this.retainMissingValue = retainMissingValue;
    this.replaceMissingValueWith = Strings.emptyToNull(replaceMissingValueWith);
```